File: day5.py
#!python

instr = [ 0, 3, 0, 1, -3]

with open('input/day5.txt') as f:
    instr = map(int, f.readlines())
    pointer = 0
    total_jumps = 0

    while True:
        this_jump = instr[pointer]
        # inst[pointer] += 1                            # easy case
        instr[pointer] += 1 if this_jump < 3 else -1    # hard case
        pointer += this_jump
        total_jumps += 1
        if pointer < 0 or pointer >= len(instr):
            print "escaped in {} jumps".format(total_jumps)
            break

File: 2017/day5.c
#include <stdio.h>
#define ARR_SIZE 1200

int main() {
    FILE *fp = fopen("input/day5.txt", "r");
    int instr[ARR_SIZE];

    int i = 0;
    int num;

    while(fscanf(fp, "%d", &num) > 0)
        instr[i++] = num;

    int len = i;
    i = 0;
    int jumps = 0;

    while (i >= 0 && i < len) {
        int this_jump = instr[i];
        instr[i] += this_jump < 3 ? 1 : -1;
        i += this_jump;
        jumps++;
    }

    printf("total jumps: %d\n", jumps);

    return 0;
}
